chainer/training/updaters/multiprocess_parallel_updater.py

```diff
@@ -157,6 +157,12 @@ def __init__(self, iterators, optimizer, converter=convert.concat_examples,
             main = devices.pop('main')
             devices = list(six.itervalues(devices))
             devices = [main] + devices
+        elif isinstance(devices, list):
```

Comment:
```
elif isinstance(devices, (list, tuple)):
    devices = list(devices)
```

Reply:
@mitmul

Reply:
Sorry for my late reaction. I fixed this!